prodigy_pdf/__init__.py

```diff
@@ -86,3 +88,109 @@ def before_db(examples):
             }
         },
     }
+
+
+def page_to_cropped_image(pil_page: Image, span: Dict, scale: int):
+    left, upper = span['x'], span['y']
+    right, lower = left + span['width'], upper + span['height']
+    scaled = (left * scale, upper * scale, right * scale, lower * scale)
+    cropped = pil_page.crop(scaled)
+    with BytesIO() as buffered:
+        cropped.save(buffered, format="JPEG")
+        img_str = base64.b64encode(buffered.getvalue())
+    return cropped, f"data:image/png;base64,{img_str.decode('utf-8')}"
+
+
+def fold_ocr_dashes(ocr_input:str) -> str:
+    """
+    OCR might literally add dashes at the end of the line to indicate
+    continuation of the word. This can be fine in some cases, but this
+    function can fold it all into a single string. 
+    """
+    new = ""
+    for line in ocr_input.split("\n"):
+        line = line.strip()
+        if line.rfind("-") == -1:
+            newline = line + " "
+        else:
+            newline = line[:line.rfind("-")]
+        new += newline
+    return new.strip()
+
+
+def _validate_ocr_example(ex: Dict):
+    if 'meta' not in ex:
+        raise ValueError(f"It seems the `meta` key is missing from an example: {ex}. Did you annotate this data with `pdf.image.manual`?")
+    if 'path' not in ex['meta']:
+        raise ValueError(f"It seems the `path` key is missing from an example metadata: {ex}. Did you annotate this data with `pdf.image.manual`?")
+    if 'page' not in ex['meta']:
+        raise ValueError(f"It seems the `page` key is missing from an example metadata: {ex}. Did you annotate this data with `pdf.image.manual`?")
+
+
+@recipe(
+    "pdf.ocr.correct",
+    # fmt: off
+    dataset=("Dataset to save answers to", "positional", None, str),
+    source=("Source with PDF Annotations", "positional", None, str),
+    labels=("Labels to consider", "option", "l", str),
+    scale=("Zoom scale. Increase above 3 to upscale the image for OCR.", "option", "s", int),
+    remove_base64=("Remove base64-encoded image data", "flag", "R", bool),
+    fold_dashes=("Removes dashes at the end of a textline and folds them with the next term.", "flag", "f", bool),
+    autofocus=("Autofocus on the transcript UI", "flag", "af", bool)
+    # fmt: on
+)
+def pdf_ocr_correct(
+    dataset: str,
+    source: str,
+    labels: str,
+    scale: int = 3,
+    remove_base64:bool=False,
+    fold_dashes:bool = False,
+    autofocus: bool = False
+) -> ControllerComponentsDict:
+    """Applies OCR to annotated segments and gives a textbox for corrections."""
+    stream = get_stream(source)
+    labels = labels.split(",")
```

Comment:
Nit, but we have a type mismatch. Labels are first str and then they become a list.